### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by akallabeth**
* Cleans up a lot of warnings
* Removes `strlen` and `strcpy` where possible.

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/core/nego.c [hunk 1]
```diff
@@ -745,9 +746,10 @@ static BOOL nego_read_request_token_or_cookie(rdpNego* nego, wStream* s)
 	{
 		Stream_Rewind(s, 2);
 		len = Stream_GetPosition(s) - pos;
+		remain = Stream_GetRemainingLength(s);
 		Stream_Write_UINT16(s, 0);
 
-		if (strlen((char*)str) == len)
+		if (strnlen((char*)str, remain) == len)
```
Comment (mfleisz) [inline libfreerdp/core/nego.c]: I think this should be `strnlen(str, len)`

File: libfreerdp/crypto/crypto.c [hunk 2]
```diff
@@ -741,7 +741,7 @@ char** crypto_cert_get_dns_names(X509* x509, int* count, int** lengths)
 	for (i = 0; i < list.count; i ++)
 	{
 		result[i] = list.strings[i];
-		(*lengths)[i] = strlen(result[i]);
+		(*lengths)[i] = strnlen(result[i], sizeof(*result));
```
Comment (mfleisz) [inline libfreerdp/crypto/crypto.c]: Think `result[i]` can be longer than `sizeof(char*)`?

**Comment by akallabeth** [comment]
@freerdp-bot test

**Pull request merged**